packages/firestore/src/local/indexeddb_query_cache.ts

```diff
@@ -297,6 +298,8 @@ export class IndexedDbQueryCache implements QueryCache {
     this.garbageCollector = gc;
   }
 
+  // TODO(gsoltis): we can let the compiler assert that txn !== null if we
+  // drop null from the type bounds on txn.
```

Comment:
Well, kinda. It compiles, but only because we don't enforce strict function types. This is replicating the type signature from `GarbageSource`, which allows `null`. So actually removing it from this function gets us no safety and a more obscure runtime error than what you would currently get. 

The solution will eventually arrive when we get rid of `GarbageSource`.